source/include/ice_api.h

```diff
@@ -0,0 +1,105 @@
+                                   uint32_t passwordLen,
+                                   uint32_t * pStunMessageBufferLength );
+
+IceResult_t Ice_CreateRequestForSrflxCandidate( IceAgent_t * pIceAgent,
+                                                uint8_t * pStunMessageBuffer,
+                                                uint8_t * pTransactionIdBuffer,
+                                                uint32_t * pSendStunMessageBufferLength );
+
+IceResult_t Ice_CreateRequestForNominatingValidCandidatePair( IceAgent_t * pIceAgent,
+                                                              uint8_t ** ppSendStunMessageBuffer,
+                                                              uint32_t * pSendStunMessageBufferLength,
+                                                              IceCandidatePair_t * pIceCandidatePair,
+                                                              uint8_t * pTransactionIdBuffer );
+
+IceResult_t Ice_CreateRequestForConnectivityCheck( IceAgent_t * pIceAgent,
+                                                   uint8_t ** ppSendStunMessageBuffer,
+                                                   uint32_t * pSendStunMessageBufferLength,
+                                                   uint8_t * pTransactionIdBuffer,
+                                                   IceCandidatePair_t * pIceCandidatePair );
+
+IceResult_t Ice_CreateResponseForRequest( IceAgent_t * pIceAgent,
+                                          uint8_t ** ppSendStunMessageBuffer,
+                                          uint32_t * pSendStunMessageBufferLength,
+                                          IceIPAddress_t * pSrcAddr,
+                                          uint8_t * pTransactionIdBuffer );
+
+IceResult_t Ice_DeserializeStunPacket( StunContext_t * pStunCxt,
+                                       StunHeader_t * pStunHeader,
+                                       StunAttribute_t * pStunAttribute,
+                                       IceStunDeserializedPacketInfo_t * pDeserializedPacketInfo );
+
+IceStunPacketHandleResult_t Ice_HandleStunPacket( IceAgent_t * pIceAgent,
+                                                  uint8_t * pReceivedStunMessageBuffer,
+                                                  uint32_t receivedStunMessageBufferLength,
+                                                  uint8_t ** ppSendTransactionIdBuffer,
+                                                  uint8_t ** ppSendStunMessageBuffer,
+                                                  uint32_t * pSendStunMessageBufferLength,
+                                                  IceIPAddress_t * pLocalCandidateAddress,
+                                                  IceIPAddress_t * pRemoteCandidateAddress,
+                                                  IceCandidatePair_t ** ppIceCandidatePair );
+
+IceResult_t Ice_HandleServerReflexiveCandidateResponse( IceAgent_t * pIceAgent,
```

Comment:
Same here. Can we move this to other header file to avoid user including this unexpectly?

Reply:
updated.